File: web/gen-icons.py
from PIL import Image, ImageDraw
import math

def make_icon(size):
    img = Image.new('RGBA', (size, size), (10, 10, 10, 255))
    draw = ImageDraw.Draw(img)
    cx, cy = size / 2, size / 2
    r = size * 0.28
    # circle
    draw.ellipse([cx - r, cy - r, cx + r, cy + r], outline=(255, 255, 255, 30), width=max(2, int(size * 0.02)))
    # hands
    lw = max(2, int(size * 0.015))
    draw.line([cx, cy, cx, cy - r * 0.6], fill=(255, 255, 255, 110), width=lw)
    draw.line([cx, cy, cx + r * 0.45, cy], fill=(255, 255, 255, 110), width=lw)
    # center dot
    dot_r = max(3, size * 0.035)
    draw.ellipse([cx - dot_r, cy - dot_r, cx + dot_r, cy + dot_r], fill=(74, 222, 128, 160))
    # rounded corners via mask
    rad = int(size * 0.18)
    mask = Image.new('L', (size, size), 0)
    md = ImageDraw.Draw(mask)
    md.rounded_rectangle([0, 0, size - 1, size - 1], rad, fill=255)
    img.putalpha(mask)
    img.save(f'icon-{size}.png')

make_icon(192)
make_icon(512)
print('icons done')

File: web/gen-icons.js
const { createCanvas } = require('canvas');
const fs = require('fs');

function makeIcon(size) {
  const c = createCanvas(size, size);
  const ctx = c.getContext('2d');
  // dark bg
  ctx.fillStyle = '#0a0a0a';
  ctx.beginPath();
  ctx.roundRect(0, 0, size, size, size * 0.2);
  ctx.fill();
  // center dot/clock
  const cx = size / 2, cy = size / 2, r = size * 0.3;
  ctx.strokeStyle = 'rgba(255,255,255,.15)';
  ctx.lineWidth = size * 0.02;
  ctx.beginPath();
  ctx.arc(cx, cy, r, 0, Math.PI * 2);
  ctx.stroke();
  // hands
  ctx.strokeStyle = 'rgba(255,255,255,.5)';
  ctx.lineWidth = size * 0.015;
  ctx.beginPath(); ctx.moveTo(cx, cy); ctx.lineTo(cx, cy - r * 0.65); ctx.stroke();
  ctx.beginPath(); ctx.moveTo(cx, cy); ctx.lineTo(cx + r * 0.5, cy); ctx.stroke();
  // center dot
  ctx.fillStyle = 'rgba(74,222,128,.7)';
  ctx.beginPath(); ctx.arc(cx, cy, size * 0.04, 0, Math.PI * 2); ctx.fill();

  fs.writeFileSync(`icon-${size}.png`, c.toBuffer('image/png'));
}

makeIcon(192);
makeIcon(512);
console.log('icons done');
